linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala

```diff
@@ -111,12 +111,12 @@ object ResponseClassifiers {
 
   def HeaderRetryable(classifier: ResponseClassifier): ResponseClassifier =
     ResponseClassifier.named(s"HeaderRetryable[$classifier]") {
-      case rr@ReqRep(req, Return(rsp: Response)) if classifier.isDefinedAt(rr) =>
+      case rr if classifier.isDefinedAt(rr) =>
         val rc = classifier(rr)
-        if (rc == ResponseClass.NonRetryableFailure && Headers.Retryable.get(rsp.headerMap)) {
-          ResponseClass.RetryableFailure
-        } else {
-          rc
+        rr match {
+          case ReqRep(req, Return(rsp: Response)) if rc == ResponseClass.NonRetryableFailure && Headers.Retryable.get(rsp.headerMap) =>
+            ResponseClass.RetryableFailure
+          case _ => rc
         }
```

Comment:
i'd check rc before destructuring...

```scala
classifier(rr) match {
  ResponseClass.NonRetryableFailure =>
    rr match {
      case ReqRep(req, Return(rsp)) if Headers.Retryable.get(rsp.headerMap) => ResponseClass.RetryableFailure
      case _=> rc
    }
    case _ => rc
}
```

this should be a bit less work in the common case